### PR TITLE
Checkbox: Updates to look and feel and using new theme

### PR DESCRIPTION
**Pull request by torkelo**
* Updates checkbox to new design based on figma and new theme
* Updates Label component to use new theme
* Updates annotation switch to use InlineSwitch instead of legacy switch

Screenshot from 2021-04-23 12-02-12
Screenshot from 2021-04-23 12-01-46

### CODE REVIEW

File: packages/grafana-ui/src/components/TimePicker/TimeZonePicker/TimeZoneOption.tsx [hunk 1]
```diff
@@ -104,34 +102,26 @@ const getStyles = stylesFactory((theme: GrafanaTheme) => {
...
       &:hover {
-        background: ${theme.colors.dropdownOptionHoverBg};
+        background: ${theme.colors.action.hover};
 
-        span.${offsetClassName} {
-          background: ${offsetHoverBg};
-        }
+        // span.${offsetClassName} {
+        //   background: ${theme.colors.action.hover};
+        // }
```
Comment (mckn) [inline packages/grafana-ui/src/components/TimePicker/TimeZonePicker/TimeZoneOption.tsx:112]: nit: should this be removed?

File: packages/grafana-ui/src/components/TimePicker/TimeZonePicker/TimeZoneOption.tsx [hunk 2]
```diff
@@ -104,34 +102,26 @@ const getStyles = stylesFactory((theme: GrafanaTheme) => {
...
-      border-bottom: 0;
-      border-left-width: 2px;
+      background: ${theme.colors.action.hover};
 
-      span.${offsetClassName} {
-        background: ${offsetHoverBg};
-      }
+      // span.${offsetClassName} {
+      //   background: ${theme.colors.action.hover};
+      // }
```
Comment (mckn) [inline packages/grafana-ui/src/components/TimePicker/TimeZonePicker/TimeZoneOption.tsx:120]: nit: should this be removed?

**Review by mckn — APPROVED** [review]
Two small nits but looks good to me!

File: packages/grafana-ui/src/components/TimePicker/TimeZonePicker/TimeZoneOption.tsx (hunk 1 above)
Comment (torkelo) [reply packages/grafana-ui/src/components/TimePicker/TimeZonePicker/TimeZoneOption.tsx:112]: oh, yea, was going to test firs how it looked and if this was needed, forgot about it :)

**Pull request merged**